spec/message.abnf

```diff
@@ -1,43 +1,46 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = simple-start-char / text-escape / expression
+pattern = *(text-char / text-escape / expression)
 
-declaration = input-declaration / local-declaration
+complex-message = *(declaration [s]) complex-body [s]
+declaration = input-declaration / local-declaration / reserved-keyword
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 
-body = quoted-pattern
-     / (selectors 1*([s] variant))
-
+complex-body = quoted-pattern
+             / (selectors 1*([s] variant))
```

Comment:
Just name this `body` like it was before.

I think this also needs `reserved-keyword`? So perhaps:

```suggestion
body = quoted-pattern
     / (selectors 1*([s] variant))
     / reserved
```

Reply:
If we're taking the "body" term exclusively for complex messages, then we should rename `reserved-body` to something else. My preference would be to instead specify it as `complex-body`, because that's where it's used.

As we're not allowing any `{}` nesting in reserved syntax, I don't see how we'd later allow for it replacing the whole of a complex message's body. Could you describe what sort of future extension you had in mind here?

Reply:
> Could you describe what sort of future extension you had in mind here?

Basically, a different type of selector from `.match`

Reply:
For that, we need something like this:

```suggestion
complex-body = quoted-pattern
             / ((selectors / reserved-statement) 1*([s] variant))
```

Feel free to commit that suggestion to this PR if you agree.

Reply:
Actually, let's not change this line. The `reserved-statement` alternative should really go within `selectors` instead.